src/Core/Settings.cpp

```diff
@@ -3635,7 +3635,7 @@ Allows to ignore errors for MATERIALIZED VIEW, and deliver original block to the
 Ignore MVs with dropped target table during pushing to views
 )", 0) \
     DECLARE(Bool, allow_materialized_view_with_bad_select, true, R"(
-Allow CREATE MATERIALIZED VIEW with SELECT query that references nonexistent tables or columns. It must still be syntactically valid. Doesn't apply to refreshable MVs. Doesn't apply if the MV schema needs to be inferred from the SELECT query (i.e. if the CREATE has no column list and no TO table). Can be used for creating MV before its source table.
```

Comment:
it is still possible to create mv from unknown table but only if you don't specify a TO table. https://fiddle.clickhouse.com/982b7567-452b-4007-9d19-521b50351d02

Reply:
it's not related to `TO` it's related to columns inference https://fiddle.clickhouse.com/6ba9e6fa-a149-4422-93d0-5a3d07e50030

Reply:
Good point. Maybe it's easiest to leave it as is.